[Enhancement] remove delete restriction on the auto-increment table(#20702)

srlch opened the pull request:
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #20702

## Problem Summary(Required) ：
In current implementation, when the auto increment column is value column but not the key column,
this kind of table can not support delete. The pr will remove restriction, but has some side affect as following:

1. If the delete-partial update conflict happen, partial update operation maybe lost.
2. If it is the streamload combine with the delete and partial update ops and manipulate on a row which has existed
    in the previous version, all the partial update ops after delete ops maybe lost for this row if they contained in
    different segment file.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

decster commented on gensrc/proto/internal_service.proto:
```diff
@@ -169,12 +169,11 @@ message PTabletWriterOpenRequest {
-    optional int32 sender_id = 30;
-    optional PartialUpdateMode partial_update_mode = 31;
+    optional bool is_incremental = 28 [default = false];
+    optional int32 sender_id = 29;
+    optional PartialUpdateMode partial_update_mode = 30;
```
change field id may not be compactible, does 3.0 have release already?

srlch committed:
[Enhancement] remove delete restriction on the auto-increment table(#20702)

In current implementation, when the auto increment column is value column but not the key column,
this kind of table can not support delete. The pr will remove restriction, but has some side affect as following:

1. If the delete-partial update conflict happen, partial update operation maybe lost.
2. If it is the streamload combine with the delete and partial update ops and manipulate on a row which has existed
   in the previous version, all the partial update ops after delete ops maybe lost for this row if they contained in
   different segment file.

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
@Mergifyio backport branch-3.0
